lib/query.js

```diff
@@ -147,22 +147,17 @@ Query.prototype.handleError = function (err, connection) {
 
 Query.prototype.submit = function (connection) {
   if (typeof this.text !== 'string' && typeof this.name !== 'string') {
-    const err = new Error('A query must have either text or a name. Supplying neither is unsupported.')
-    connection.emit('error', err)
-    connection.emit('readyForQuery')
-    return
+    return new Error('A query must have either text or a name. Supplying neither is unsupported.')
```

Comment:
Returning an error doesn't seem like a widely-used practice. Why are we not throwing it?

Reply:
It would have to be caught to be passed to the callback and it wouldn’t be easy to distinguish from other errors at that point.

Reply:
Now that you mention it, do we need to distinguish it from other errors? Reporting an error that occurred in the stack when calling Query.submit as a "query error" sounds rather intuitive.

Reply:
There shouldn’t be any other errors so I prefer not to treat those like they’re expected if they do occur.